majifix311/src/main/java/com/example/majifix311/ui/auth/SigninActivity.java

```diff
@@ -0,0 +1,422 @@
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Predicate;
+import io.reactivex.observers.DisposableObserver;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.adapter.rxjava2.HttpException;
+
+/**
+ * Reactive Signin activity
+ *
+ * @see <a href="https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/">https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/</a>
+ * @see <a href="https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572">https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572</a>
+ */
+public class SigninActivity extends AppCompatActivity {
+
+    public static final String TAG = SigninActivity.class.getSimpleName();
+
+    //view references
+    private TextInputLayout emailInputLayout;
+    private TextInputLayout passwordInputLayout;
+    private TextInputEditText emailTextInput;
+    private TextInputEditText passwordTextInput;
+    private Button signInButton;
+
+    //rxbindings
+    private CompositeDisposable compositeDisposable;
+    private Observable<CharSequence> emailObservable;
+    private Observable<CharSequence> passwordObservable;
+    private Observable<Object> signInButtonObservable;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signin);
+
+        //ensure auth initialization
+        Auth.init(getApplicationContext(), BuildConfig.END_POINT);
+
+        //obtain view
+        emailInputLayout = (TextInputLayout) findViewById(R.id.auth_til_email);
+        passwordInputLayout = (TextInputLayout) findViewById(R.id.auth_til_password);
+        emailTextInput = (TextInputEditText) findViewById(R.id.auth_tiedt_email);
+        passwordTextInput = (TextInputEditText) findViewById(R.id.auth_tiedt_password);
+        signInButton = (Button) findViewById(R.id.auth_btn_signin);
+
+        //rx init
+        compositeDisposable = new CompositeDisposable();
+        observe();
+        subscribe();
+    }
+
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        compositeDisposable.dispose();
+    }
+
+
+    /**
+     * init observables
+     */
+    private void observe() {
+        emailObservable = RxTextView.textChanges(emailTextInput);
+        passwordObservable = RxTextView.textChanges(passwordTextInput);
+        signInButtonObservable = RxView.clicks(signInButton);
+    }
+
+    /**
+     * init subscription
+     */
+    private void subscribe() {
+
+        //observe email text input changes
+        Disposable emailDisposable = emailObservable
+                .doOnNext(new Consumer<CharSequence>() {
+                    @Override
+                    public void accept(CharSequence charSequence) throws Exception {
+                        hideEmailValidationError();
```

Comment:
This causes the error to "jump" with each character insert. Why are you clearing here as well as after verification?